dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/metadata/MetadataPresenter.java

```diff
@@ -50,102 +52,109 @@ public void initialize(final URL location, final ResourceBundle resources) {
-                            return new TableCell<MetadataEntry, String>() {
-                                @Override
-                                protected void updateItem(final String item, final boolean empty) {
-                                    if (item != null) {
-                                        super.updateItem(item, empty);
-                                        setText(item);
-                                        setTooltip(new Tooltip(item));
+                                new TableCell<MetadataEntry, String>() {
+                                    @Override
+                                    protected void updateItem(final String item, final boolean empty) {
+                                        if (item != null) {
+                                            super.updateItem(item, empty);
+                                            setText(item);
+                                            setTooltip(new Tooltip(item));
+                                        }
                                     }
-                                }
-                            };
-                        });
+                                });
     }
 
-    //create meta data keys for showing on server
-    private ImmutableList.Builder<MetadataEntry> createMetaDataBuilder(final Metadata metadata, final ImmutableList.Builder<MetadataEntry> builder) {
+    private void initLabels() {
+        metadataTableColName.setText(resourceBundle.getString("metadataTableColName"));
+        metadataTableColValue.setText(resourceBundle.getString("metadataTableColValue"));
+        objectName.setText(ds3Metadata.getName());
+        objectSize.setText(ByteFormat.humanReadableByteCount(ds3Metadata.getSize(), false));
+        lastModified.setText(ds3Metadata.getLastModified());
+        nameTooltip.setText(ds3Metadata.getName());
+    }
+
+    //create metadata keys for showing on server
+    public ImmutableList.Builder<MetadataEntry> createMetadataBuilder(final Metadata metadata, final ImmutableList.Builder<MetadataEntry> builder) {
         try {
-            //if meta data does not contains creation time key thens show all metadata got from bp server without any processing
-            if (metadata.get(creationTimeKey).size() > 0) {
+            //if metadata does not contains creation time key then show all metadata got from bp server without any processing
+            if (metadata.get(StringConstants.CREATION_TIME_KEY).size() > 0) {
                 //get the creation time from server
-                String creationTime = metadata.get(creationTimeKey).stream().findFirst().orElse(null);
-                if (creationTime.contains("T")) {
-                    creationTime = creationTime.replace("T", " ");
-                    builder.add(new MetadataEntry(creationTimeKey, creationTime));
+                String creationTime = metadata.get(StringConstants.CREATION_TIME_KEY).stream().findFirst().orElse(null);
```

Comment:
Split the logic that figures out the creation date into it's own method so that this can be updated to:

```java

final String creationDate = getCreationDate(...);
```